### PR TITLE
Use real credentials in the deployment flow

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- set your guimaas settings
- use the df flag
- load the deployment flow
- choose a cloud
- if you have creds in JEM then you should get the form to select your cred and region
- if not you should get the add credential form
- if you have creds then click on the creds select box
- click on "Add credentail..."
- you should now be shown the form to add credentials
- if you're able to add regions use the CLI then you can test add credentials (this might not currently be working)

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5496/
Test PASSed.

File: jujugui/static/gui/src/app/components/deployment-flow/credential/credential.js [hunk 1]
```diff
@@ -23,12 +23,117 @@ YUI.add('deployment-credential', function() {
...
+      setCredential: React.PropTypes.func.isRequired,
+      setRegion: React.PropTypes.func.isRequired,
+      setTemplate: React.PropTypes.func.isRequired,
+      template: React.PropTypes.string,
+      users: React.PropTypes.object.isRequired,
+      validateForm: React.PropTypes.func.isRequired
     },
 
+    credentialXHR: null,
+    regionsXHR: null,
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/credential.js]: These two will need to move into the `getInitialState` method as properties defined here are shared across all instances of this component. While here it's not really a problem, it's a good practice to get into. It really belongs in an initializer but this `getInitialState` is the closest we get without using es6 classes.

**Comment by hatched** [comment]
:shipit: I'm shipping this to avoid conflicts but we should still do a follow-up for the above comment.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**